src/ReactiveDomain.Foundation/StreamStore/ReadThroughAggregateCache.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace ReactiveDomain.Foundation.StreamStore {
+    /// <summary>
+    /// This implementation assumes that the cached aggregate may have changes since it was last seen
+    ///
+    /// Therefore the aggregate is cached on first retrieve and the store is checked for updates upon
+    /// each subsequent retrieval
+    ///
+    /// Also retrieving an aggregate at anything other than latest is disabled
+    ///
+    /// Cache management (e.g. eviction) is the responsibility of the caller/external owner
```

Comment:
How would this work? The cache itself is private, so how would the caller or external owner manage it?